recipe/patches/0001-unvendor-c-blosc.patch

```diff
@@ -0,0 +1,151 @@
+-        blosc_sources += [f for f in glob('c-blosc/blosc/*.c') if 'avx2' in f]
+-        if os.name == 'nt':
+-            define_macros += [('__AVX2__', 1)]
+-    else:
+-        info('compiling Blosc extension without AVX2 support')
++    include_dirs = [os.path.join(sys.prefix, 'include')]
++    library_dirs = [os.path.join(sys.prefix, 'lib')]
++    libraries = ["blosc"]
+ 
+     sources = ['numcodecs/blosc.pyx']
+ 
+     # define extension module
+     extensions = [
+         Extension('numcodecs.blosc',
+-                  sources=sources + blosc_sources,
++                  sources=sources,
+                   include_dirs=include_dirs,
+                   define_macros=define_macros,
+                   extra_compile_args=extra_compile_args,
++                  library_dirs=library_dirs,
++                  libraries=libraries
+                   ),
+     ]
+ 
+@@ -115,31 +75,23 @@ def blosc_extension():
+ def zstd_extension():
+     info('setting up Zstandard extension')
+ 
+-    zstd_sources = []
+     extra_compile_args = base_compile_args.copy()
+-    include_dirs = []
+     define_macros = []
+-
+-    # setup sources - use zstd bundled in blosc
+-    zstd_sources += glob('c-blosc/internal-complibs/zstd*/common/*.c')
+-    zstd_sources += glob('c-blosc/internal-complibs/zstd*/compress/*.c')
+-    zstd_sources += glob('c-blosc/internal-complibs/zstd*/decompress/*.c')
+-    zstd_sources += glob('c-blosc/internal-complibs/zstd*/dictBuilder/*.c')
+-    include_dirs += [d for d in glob('c-blosc/internal-complibs/zstd*')
+-                     if os.path.isdir(d)]
+-    include_dirs += [d for d in glob('c-blosc/internal-complibs/zstd*/*')
+-                     if os.path.isdir(d)]
+-    # define_macros += [('CYTHON_TRACE', '1')]
++    include_dirs = [os.path.join(sys.prefix, 'include')]
++    library_dirs = [os.path.join(sys.prefix, 'lib')]
++    libraries = ["zstd"]
+ 
+     sources = ['numcodecs/zstd.pyx']
+ 
+     # define extension module
+     extensions = [
+         Extension('numcodecs.zstd',
+-                  sources=sources + zstd_sources,
++                  sources=sources,
+                   include_dirs=include_dirs,
+                   define_macros=define_macros,
+                   extra_compile_args=extra_compile_args,
++                  library_dirs=library_dirs,
++                  libraries=libraries
+                   ),
+     ]
+ 
+@@ -151,22 +103,21 @@ def lz4_extension():
+ 
+     extra_compile_args = base_compile_args.copy()
+     define_macros = []
+-
+-    # setup sources - use LZ4 bundled in blosc
+-    lz4_sources = glob('c-blosc/internal-complibs/lz4*/*.c')
+-    include_dirs = [d for d in glob('c-blosc/internal-complibs/lz4*') if os.path.isdir(d)]
+-    include_dirs += ['numcodecs']
+-    # define_macros += [('CYTHON_TRACE', '1')]
++    include_dirs = [os.path.join(sys.prefix, 'include')]
++    library_dirs = [os.path.join(sys.prefix, 'lib')]
++    libraries = ["lz4"]
```

Comment:
An error on `win64`:
```
"C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\link.exe" /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:C:\b\abs_fd17djrz3t\croot\numcodecs_1707343355028\_h_env\lib /LIBPATH:C:\b\abs_fd17djrz3t\croot\numcodecs_1707343355028\_h_env\libs /LIBPATH:C:\b\abs_fd17djrz3t\croot\numcodecs_1707343355028\_h_env /LIBPATH:C:\b\abs_fd17djrz3t\croot\numcodecs_1707343355028\_h_env\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\ATLMFC\lib\x64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64" /LIBPATH:C:\b\abs_fd17djrz3t\croot\numcodecs_1707343355028\_h_env\Library\lib /LIBPATH:C:\b\abs_fd17djrz3t\croot\numcodecs_1707343355028\_h_env\Library\lib lz4.lib /EXPORT:PyInit_lz4 build\temp.win-amd64-cpython-39\Release\numcodecs/lz4.obj /OUT:build\lib.win-amd64-cpython-39\numcodecs\lz4.cp39-win_amd64.pyd /IMPLIB:build\temp.win-amd64-cpython-39\Release\numcodecs\lz4.cp39-win_amd64.lib

  LINK : fatal error LNK1181: cannot open input file 'lz4.lib'

  [numcodecs] command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64\\link.exe' failed with exit code 1181
``` 

@cbouss @SC426 Looks like the name of the library is `liblz4.lib` on Windows, see https://github.com/AnacondaRecipes/lz4-c-feedstock/blob/359bebe42162aacde2c674faaef7bc52a2ce74c7/recipe/meta.yaml#L55 and not `lz4.lib`

Reply:
yeah, that's the name in the package as well. NB windows gives "skipped" when it really means "failure" at the moment.

Reply:
can you just set the libraries list dependent on the operating system using an if/else statement?

Reply:
I updated the patch. Kind of concerning to see the build not to be marked as failed...